akka-cluster/src/main/scala/akka/cluster/sbr/DowningStrategy.scala

```diff
@@ -317,17 +317,10 @@ import akka.coordination.lease.scaladsl.Lease
     unreachableMembers.forall(m => m.status == MemberStatus.Down || m.status == MemberStatus.Exiting)
   }
 
-  def reverseDecision(decision: Decision): Decision = {
+  def reverseDecision(decision: AcquireLeaseDecision): Decision = {
```

Comment:
That's the main change. That method is only called when `Decision` is already narrow down to `AcquireLeaseDecision`. We don't need pattern match on the others.

All the reversing logic is useless except for `AcquireLeaseAndDownUnreachable` and `AcquireLeaseAndDownIndirectlyConnected`.

Reply:
Ah, I didn't get that when you commented on it, nice.